Documentation/proposals/202405-scrapeconfig-graduation.md

```diff
@@ -0,0 +1,176 @@
+
+## Goals
+
+- Enhance the Service Discovery support by adding more Service Discovery configurations.
+- Ensure all fields in Prometheus `scrape_config` are represented in the ScrapeConfig CRD.
+- Maintain naming conventions and field consistency across existing and new Service Discoveries.
+- Identify inconsistencies in validations.
+- Pave the way for graduating the ScrapeConfig CRD to Beta.
+
+## Audience
+
+- Prometheus Operator maintainers and contributors.
+- Users and developers relying on the Prometheus Operator for monitoring configurations.
+- Stakeholders interested in the evolution and improvement of the Prometheus Operator.
+
+## Non-Goals
+
+- Implementing all 28 Service Discovery configurations immediately.
+- To plan out a detailed graduation strategy.
+
+## How
+
+### Enhancing Service Discovery Support
+
+In order to provide a more comprehensive and versatile monitoring solution, enhancing the
+Service Discovery support in the Prometheus Operator is crucial. This will address current
+limitations and better serve diverse user needs. The following steps outline our approach to achieving this goal:
+
+#### Additional Service Discovery Configurations
+
+By adding more Service Discovery configurations, we increase the flexibility and utility of the CRD for various user scenarios.
+We believe that most of the user's needs can be solved with the Kubernetes service discovery and the other existing service discoveries that we offer
+and so implementing all the Service discoveries is not a priority. However, this does not imply service discovery support is halted at this number.
+It would be a better use of time to add the service discoveries that the users need as we get feature requests for them. As a starting point,
+we shall add the following 3 Service discoveries, the choice for which is not necessarily backed up by any usage data.
+
+- **`dockerswarm_sd_config`**: Docker Swarm is a popular container orchestration tool. Supporting Docker Swarm Service Discovery will cater to users leveraging Docker Swarm for their deployments.
+- **`puppetdb_sd_config`**: PuppetDB is a storage service that keeps data generated by Puppet. Integrating PuppetDB Service Discovery will be beneficial for users who utilize Puppet for configuration management.
+- **`lightsail_sd_config`**: Amazon Lightsail is a simplified cloud platform. Including Lightsail Service Discovery will assist users who prefer Lightsail for its ease of use and cost-effectiveness.
+
+The following is a list of Service Discoveries we will not be supporting yet:
+*Note: this list is subject to change and the Service Discoveries can be added on respective feature requests. Users
+are encouraged to file a feature request if they see a service discovery they use in the list.*
+
+- **`ovhcloud_sd_config`**
+- **`ionos_sd_config`**
+- **`marathon_sd_config`**
+- **`nerve_sd_config`**
+- **`serverset_sd_config`**
+- **`triton_sd_config`**
```

Comment:
I'd move Marathon, Nerve, ServerSet and Triton at the bottom of the list and highlight that these environments are either deprecated, inactive or not widespread enough that we need to care about them.